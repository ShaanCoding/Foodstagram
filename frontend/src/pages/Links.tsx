import { Link } from 'react-router-dom'

const Links = () => {
	return (
		<div className="w-3/5 ml-auto mr-auto">
			<h1 className="text-center text-xl">Feature Links</h1>
			<ul className="list-disc">
				<li>
					<Link to="/login">Login (Jack)</Link>
				</li>
				<li>
					<Link to="/register">Registration (Jack)</Link>
				</li>
				<li>
					<Link to="/profile">Profile (Jabriel)</Link>
				</li>
				<li>
					<Link to="/editprofile">Edit Profile (Jabriel)</Link>
				</li>
				<li>
					<Link to="/links">Create Post (Alwin)</Link>
				</li>
				<li>
					<Link to="/feed">Feed (Brandon)</Link>
				</li>
				<li>
					<Link to="/manageposts">Managing Posts (Shaan)</Link>
				</li>
				<li>
					<Link to="/scheduleposts">Schedule Posts (Shaan)</Link>
				</li>
				<li>
					<Link to="/search">Search (Trong)</Link>
				</li>
			</ul>
		</div>
	)
}

export default Links
