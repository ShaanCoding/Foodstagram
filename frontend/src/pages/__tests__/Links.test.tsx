import { QueryClient, QueryClientProvider } from 'react-query'
import { BrowserRouter, Router } from 'react-router-dom'
import renderer from 'react-test-renderer'
import Links from '../Links'

const queryClient = new QueryClient()

it('The Links page renders correctly', () => {
	const tree = renderer
		.create(
			<QueryClientProvider client={queryClient}>
				<BrowserRouter>
					<Links />
				</BrowserRouter>
			</QueryClientProvider>
		)
		.toJSON()
	expect(tree).toMatchSnapshot()
})
