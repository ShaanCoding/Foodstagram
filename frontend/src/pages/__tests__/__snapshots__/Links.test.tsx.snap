// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Links page renders correctly 1`] = `
<div
  className="w-3/5 ml-auto mr-auto"
>
  <h1
    className="text-center text-xl"
  >
    Feature Links
  </h1>
  <ul
    className="list-disc"
  >
    <li>
      <a
        href="/login"
        onClick={[Function]}
      >
        Login (Jack)
      </a>
    </li>
    <li>
      <a
        href="/register"
        onClick={[Function]}
      >
        Registration (Jack)
      </a>
    </li>
    <li>
      <a
        href="/profile"
        onClick={[Function]}
      >
        Profile (Jibreil)
      </a>
    </li>
    <li>
      <a
        href="/editprofile"
        onClick={[Function]}
      >
        Edit Profile (Jibreil)
      </a>
    </li>
    <li>
      <a
        href="/links"
        onClick={[Function]}
      >
        Create Post (Alwin)
      </a>
    </li>
    <li>
      <a
        href="/feed"
        onClick={[Function]}
      >
        Feed (Brandon)
      </a>
    </li>
    <li>
      <a
        href="/manageposts"
        onClick={[Function]}
      >
        Managing Posts (Shaan)
      </a>
    </li>
    <li>
      <a
        href="/scheduleposts"
        onClick={[Function]}
      >
        Schedule Posts (Shaan)
      </a>
    </li>
    <li>
      <a
        href="/updateposts"
        onClick={[Function]}
      >
        Update Posts (Shaan)
      </a>
    </li>
    <li>
      <a
        href="/search"
        onClick={[Function]}
      >
        Search (Trong)
      </a>
    </li>
  </ul>
</div>
`;
